lib/templates.js

```diff
@@ -173,31 +177,40 @@ const createAPIResourceTemplate = (serviceName, serviceDefinition, appConfig, pa
 
             ...ordExtensions,
         };
-
-        apiResources.push(obj);
+        // visibility can also be set as an ORD extension; in this case, it overwrites the default visibility
+        if (obj.visibility === RESOURCE_VISIBILITY.public) apiResources.push(obj);
     });
 
-    if (apiResources.length > 0) return apiResources;
+    return apiResources;
 };
 
-const createEventResourceTemplate = (serviceName, eventDefinition, appConfig, packageIds) => {
-    const ordExtensions = readORDExtensions(eventDefinition);
-    return {
+/**
+ * This is a template function to create Event Resource object for Event Resource Array.
+ * There can be only one event resource per service because all events are using the same protocol, they are always Cloud Events.
+ *
+ * @param {string} serviceName The name of the service.
+ * @param {object} serviceDefinition The definition of the service
+ * @returns {Array} An single-item array of objects for the Event Resources.
+ */
+const createEventResourceTemplate = (serviceName, serviceDefinition, appConfig, packageIds) => {
+    const ordExtensions = readORDExtensions(serviceDefinition);
+    if (!!ordExtensions.visibility && ordExtensions.visibility !== RESOURCE_VISIBILITY.public) return [];
+    return [{
         ordId: `${appConfig.ordNamespace}:eventResource:${serviceName}:v1`,
         title:
-            eventDefinition["@title"] ??
-            eventDefinition["@Common.Label"] ??
+            serviceDefinition["@title"] ??
+            serviceDefinition["@Common.Label"] ??
             `ODM ${appConfig.appName.replace(/[^a-zA-Z0-9]/g, "")} Events`,
```

Comment:
@zongqichen : This is definitely a whole service and not only an event definition. That's why I've renamed it into `serviceDefinition`.